release.rb

```diff
@@ -0,0 +1,215 @@
+    die("Failed to build framework for carthage. Aborting.") unless built_carthage_framework
+
+    puts "-> Setting podspec version"
+    set_podspec_version = system("cat #{pod_spec} | grep 's.version' | grep -q \"#{version}\"")
+    if !set_podspec_version
+      updated_podspec_version = system("sed -i.backup \"s/s.version *= *\".*\"/s.version     = \"#{version}\"/g\" #{pod_spec}")
+      unless updated_podspec_version 
+        restore_podspec
+        die("Failed to update version in podspec. Aborting.")
+      end
+
+      added_podspec = system("git add #{pod_spec}")
+      unless added_podspec
+        restore_podspec
+        die("Failed to add #{pod_spec} to index. Aborting.")
+      end
+
+      pushed_updated_version = system("git commit -m \"Bumping version to #{version}\"")
+      unless pushed_updated_version
+        restore_podspec
+        die("Failed to push updated version. Aborting.")
+      end
+    else
+      puts " > Podspec already set to #{version}. Skipping."
+    end
+
+    if force
+      puts "-> Tagging version"
+      tagged_version = system("git tag -s #{version_tag} -F #{release_notes}")
+      
+      die("Failed to tag version. Aborting.") unless tagged_version
+      puts "-> Pushing tag to origin"
+      pushed_tag = system("git push origin #{version_tag}")
+      die("Failed to push tag '#{version_tag}' to origin") unless pushed_tag
+    else
+       puts "-> Tagging version (force)"
+       tagged_version = system("git tag -f -s #{version_tag} -F #{release_notes}")
+       die("Failed to tag version. Aborting.") unless tagged_version
+
+       pushed_tag = system("git push origin #{version_tag}")
+       die("Failed to push tag #{version_tag} to origin. Aborting.") unless pushed_tag
+    end
+
+    if set_podspec_version
+      `rm #{release_notes}`
+      pushed_to_origin = system("git push origin #{remote_branch}")
+      die("Failed to push to origin. Aborting.") unless pushed_to_origin
+
+      puts " > Pushed version to origin"
+    end
+
+    puts
+    puts "---------------- Released as $VERSION_TAG ----------------"  
+    puts
+    puts "Archiving carthage release..."
+    
+    archived_carthage_release = system("#{carthage} archive #{carthage_framework_name}")
+    die("Failed to archive framework for carthage. Aborting.") unless archived_carthage_release
+
+    puts
+    #puts "Pushing to pod trunk..."
+
+    #`#{pod} trunk push #{pod_spec}`
+
+    puts
+    puts "================ Finalizing the Release ================"
+    puts
+    puts " - Go to #{github_tags_url} and mark this as a release."
+    puts "   - Paste the contents of #{release_notes} into the release notes. Tweak for Github styling."
+    puts "   - Attach #{carthage_framework_name}.framework.zip to it."
+    puts " - Announce!"
+
+    `rm #{pod_spec}.backup`
+  end
+
+  def self.die(error_message)
+    raise("[ERROR]: #{error_message}")
```

Comment:
I'm not certain that raising an error is the best way to kill the task early. I'm interested to hear if anyone agrees and has a more elegant killswitch.